pkg/apis/options/load_test.go

```diff
@@ -291,15 +341,18 @@ var _ = Describe("Load", func() {
 					},
 				},
 			}),
+			// TODO (yanasega): temporary solution: optionsWithNilProvider and legacyOptionsWithNilProvider are set because Providers is internal
+			// therefore it's loaded as nil but it has some default fields that are being set in NewOptions
+			// as appose to Upstreams/Headers alpha options
```

Comment:
Long term, we want to remove the defaulting from here and create a defaulting function for various different types. This seems like a reasonable hack for the time being!

Reply:
I wonder if this is easier to do using the `NewOptions` builder, just thinking this would make it easier as we make other changes later down the line
```
optionsWithNilProvider := NewOptions()
optionsWithNilProvider.Providers = nil
```